src/drive/web/modules/upload/UploadButton.jsx

```diff
@@ -43,4 +48,16 @@ const UploadButton = ({ label, disabled, onUpload, className }) => (
   </label>
 )
 
-export default UploadButton
+const mapDispatchToProps = (dispatch, { displayedFolder, sharingState }) => ({
+  onUpload: files => {
+    dispatch(uploadFiles(files, displayedFolder.id, sharingState))
+  }
+})
+
+export default compose(
+  withSharingState,
+  connect(
+    null,
+    mapDispatchToProps
+  )
+)(UploadButton)
```

Comment:
Can we add proptypes for this `UploadButton` component ? Since we're using its `ownProps` I think this can be very useful ;)